ingestion/functions/parsing/common/python/parsing_lib/parsing_lib_test.py

```diff
@@ -295,64 +300,119 @@ def test_finalize_upload_raises_error_for_failed_request(requests_mock):
     requests_mock.register_uri(
         "PUT",
         update_upload_url,
-        exc=requests.exceptions.ConnectTimeout)
+        [{"json": {}, "status_code": 500}, {"json": {}}])
 
     try:
-        parsing_lib.finalize_upload(_SOURCE_ID, upload_id, 42, 0, {})
-    except requests.exceptions.ConnectTimeout:
+        parsing_lib.finalize_upload(_SOURCE_ID, upload_id, {}, 42, 0)
+    except RuntimeError:
+        assert requests_mock.request_history[0].url == update_upload_url
+        assert requests_mock.request_history[1].url == update_upload_url
+        assert requests_mock.request_history[-1].json(
+        ) == {"status": "ERROR", "summary": {"error": parsing_lib.UploadError.INTERNAL_ERROR.name}}
         return
 
     # We got the wrong exception or no exception, fail the test.
     assert False
 
 
-@patch('parsing_lib.parsing_lib.get_today')
+@ patch('parsing_lib.parsing_lib.get_today')
 def test_filter_cases_by_date_today(mock_today):
     from parsing_lib import parsing_lib  # Import locally to avoid superseding mock
     mock_today.return_value = datetime.datetime(2020, 6, 8)
     cases = parsing_lib.filter_cases_by_date(
         [CASE_JUNE_FIFTH],
-        {"numDaysBeforeToday": 3, "op": "EQ"})
+        {"numDaysBeforeToday": 3, "op": "EQ"},
+        "source_id", "upload_id", {})  # api_creds
     assert list(cases) == [CASE_JUNE_FIFTH]
 
 
-@patch('parsing_lib.parsing_lib.get_today')
+@ patch('parsing_lib.parsing_lib.get_today')
 def test_filter_cases_by_date_not_today(mock_today):
     from parsing_lib import parsing_lib  # Import locally to avoid superseding mock
     mock_today.return_value = datetime.datetime(2020, 10, 10)
     cases = parsing_lib.filter_cases_by_date(
         [CASE_JUNE_FIFTH],
-        {"numDaysBeforeToday": 3, "op": "EQ"})
+        {"numDaysBeforeToday": 3, "op": "EQ"},
+        "source_id", "upload_id", {})  # api_creds
     assert cases == []
 
 
-@patch('parsing_lib.parsing_lib.get_today')
+@ patch('parsing_lib.parsing_lib.get_today')
 def test_filter_cases_by_date_exactly_before_today(mock_today):
     from parsing_lib import parsing_lib  # Import locally to avoid superseding mock
     mock_today.return_value = datetime.datetime(2020, 6, 8)
     cases = parsing_lib.filter_cases_by_date(
         [CASE_JUNE_FIFTH],
-        {"numDaysBeforeToday": 3, "op": "LT"})
+        {"numDaysBeforeToday": 3, "op": "LT"},
+        "source_id", "upload_id", {})  # api_creds
     assert cases == []
 
 
-@patch('parsing_lib.parsing_lib.get_today')
+@ patch('parsing_lib.parsing_lib.get_today')
```

Comment:
Decorators shouldn't have space between @ and the name I believe is that pep8?

Reply:
Yeah, no clue. There must have been a syntax error at one point that forced this space, and it wasn't removed afterwards. Manually corrected this, after which the formatter doesn't reattempt it.